modules/wallet/update.go

```diff
@@ -219,6 +223,10 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 // ReceiveUpdatedUnconfirmedTransactions updates the wallet's unconfirmed
 // transaction set.
 func (w *Wallet) ReceiveUpdatedUnconfirmedTransactions(txns []types.Transaction, _ modules.ConsensusChange) {
+	if err := w.tg.Add(); err != nil {
+		panic(err)
+	}
+	defer w.tg.Done()
```

Comment:
The wallet only unsubscribes from the consensus and tpool after `tg.Stop` returns. In the interim, the consensus or tpool could send another update, which the wallet should gracefully reject. That's why David is advocating a `return` here instead of a `panic`.